test/dotcom_web/controllers/schedule/predictions_test.exs

```diff
@@ -25,50 +16,63 @@ defmodule DotcomWeb.ScheduleController.PredictionsTest do
     {:ok, %{conn: conn}}
   end
 
-  describe "init/1" do
-    test "defaults to using Predictions.Repo.all" do
-      assert init([]) == [predictions_fn: &Predictions.Repo.all/1]
-    end
-  end
-
   describe "call/2" do
     test "when given a date that isn't the service date, assigns no predictions", %{conn: conn} do
       conn =
         conn
         |> assign(:date, ~D[2016-12-31])
-        |> call(@opts)
+        |> assign(:origin, Faker.Pokemon.location())
+        |> call([])
+
+      assert conn.assigns[:predictions] == []
+      assert conn.assigns[:vehicle_predictions] == []
+    end
+
+    test "when there is no origin, assigns no predictions", %{conn: conn} do
+      conn =
+        conn
+        |> assign(:origin, nil)
+        |> call([])
 
       assert conn.assigns[:predictions] == []
       assert conn.assigns[:vehicle_predictions] == []
     end
 
     test "assigns predictions for a route, stop, and direction ID", %{conn: conn} do
+      expect(Predictions.Repo.Mock, :all, fn [route: "4", direction_id: "0"] ->
+        @empty
+      end)
+
       conn =
         conn
         |> assign(:origin, %Stops.Stop{id: "place-sstat"})
         |> assign(:destination, nil)
         |> assign(:route, %{id: "4"})
         |> assign(:direction_id, "0")
-        |> call(predictions_fn: fn [route: "4", direction_id: "0"] -> @empty end)
+        |> call(init())
 
       assert conn.assigns[:predictions] == []
     end
 
     test "ignores predictions which have the origin as their destination", %{conn: conn} do
-      prediction = %Predictions.Prediction{
-        time: ~N[2017-01-01T00:00:00],
-        stop: %Stops.Stop{id: "origin"},
-        trip: 1234,
-        departing?: false
-      }
+      expect(Predictions.Repo.Mock, :all, fn _ ->
+        [
+          %Predictions.Prediction{
+            time: ~N[2017-01-01T00:00:00],
+            stop: %Stops.Stop{id: "origin"},
+            trip: 1234,
+            departing?: false
+          }
+        ]
+      end)
 
       conn =
         conn
         |> assign(:origin, %Stops.Stop{id: "origin"})
         |> assign(:destination, nil)
         |> assign(:route, %{id: "4"})
         |> assign(:direction_id, "0")
-        |> call(predictions_fn: fn _ -> [prediction] end)
+        |> call(init())
```

Comment:
`init()` will just return an empty list. And `call/1` will default to an empty list. In fact `call/2` just ignores the options. So, I'm pretty sure you can just remove it.